forch/faucetizer.py

```diff
@@ -1,49 +1,182 @@
 
 
 class Faucetizer:
     """Collect Faucet information and generate ACLs"""
-    def __init__(self):
-        self._network_state = {}
+    def __init__(self, output_file):
+        self._devices = {}
+        self._base_faucet_config = None
+        self._faucet_config = None
+        self._output_file = output_file
 
     def process_network_state(self, network_state):
-        """Process network state inputs"""
-        self._network_state = proto_dict(network_state, True)
+        """Process network state input"""
+        macs = set()
+        for mac, learning in network_state.device_mac_learnings.items():
+            macs.add(mac)
+            self._process_device_learning(mac, learning)
+        for mac, behavior in network_state.device_mac_behaviors.items():
+            macs.add(mac)
+            self._process_device_behavior(mac, behavior)
+
+        self._faucetize(macs)
+
+    def _process_device_learning(self, mac, learning):
+        device = self._devices.setdefault(mac, Device())
+        device.new_learning = learning
+
+    def _process_device_behavior(self, mac, behavior):
+        device = self._devices.setdefault(mac, Device())
+        device.behavior = behavior
+
+    def process_faucet_config(self, faucet_config):
+        """Process faucet config when base faucet config changes"""
+        self._base_faucet_config = faucet_config
+        self._faucet_config = faucet_config
+        self._faucetize()
+
+    def _faucetize(self, macs=None):
+        if not self._faucet_config:
+            return
+
+        if not macs:
+            macs = self._devices.keys()
+
+        commit = False
+
+        for mac in macs:
+            device = self._devices[mac]
 
-        sorted_rules = list(self._network_state.get("named_match_rules", {}).keys())
-        sorted_rules.sort()
-        sys.stdout.write(f'{sorted_rules}\n')
+            if device.has_migrated():
+                self._reset_port_config(device)
+                commit = True
 
-        sorted_macs = list(self._network_state.get('device_mac_behaviors', {}).keys())
-        sorted_macs.sort()
-        sys.stdout.write(f'{sorted_macs}\n')
+            if device.is_fulfilled():
+                switch = device.new_learning.switch
+                port = device.new_learning.port
+                switch_config = self._faucet_config.get('dps', {}).get(switch, {})
+                port_config = switch_config.get('interfaces', {}).get(port)
+                if not port_config:
+                    LOGGER.warning('Switch or port not defined in faucet config: %s %s',
+                                   switch, port)
+                    continue
+                port_config['native_vlan'] = device.behavior.vid
+                role = device.behavior.role
+                if role:
+                    port_config['acls_in'] = [f'role_{role}']
+                commit = True
 
+        if commit:
+            self._commit_faucet_config(macs)
```

Comment:
then pass in new config to be committed here as function argument

Reply:
Done